### PR TITLE
Resolve Sidekiq Feedback

### LINKED ISSUE #19: Meandering feedback
I'm just trying to read through this and make sense of things. I have some questions and will use this issue as a place to have a discussion.

### PR DESCRIPTION
**Pull request by jherdman**
Resolves #19

### CODE REVIEW

File: lib/active_jorb/job.ex [hunk 1]
```diff
@@ -11,20 +11,20 @@ defmodule ActiveJorb.Job do
...
   defstruct job_class: nil,
             queue_name: "default",
-            priority: "",
+            priority: nil,
             arguments: [],
             locale: "en",
-            retry: false,
-            executions: 0
+            executions: 0,
+            job_id: ""
```
Comment (elucid) [inline lib/active_jorb/job.ex]: IMO this should be `nil`

File: lib/active_jorb/job.ex [hunk 2]
```diff
@@ -11,20 +11,20 @@ defmodule ActiveJorb.Job do
...
           job_class: String.t(),
           queue_name: String.t(),
-          priority: String.t(),
+          priority: nil | String.t(),
           arguments: list(),
           locale: String.t(),
-          retry: boolean,
-          executions: pos_integer()
+          executions: pos_integer(),
+          job_id: String.t()
```
Comment (elucid) [inline lib/active_jorb/job.ex]: And this should be `nil | String.t()`

**Review by elucid — CHANGES_REQUESTED** [review]

File: lib/active_jorb/job.ex (hunk 2 above)
Comment (jherdman) [reply lib/active_jorb/job.ex]: Done!

**Comment by jherdman** [comment]
@elucid should be fixed

**Comment by elucid** [comment]
Hey @jherdman sorry for taking so long to get to this. I just took a look and the payload is missing `job_id`. It looks like the above only modifies what `ActiveJorb.QueueAdapter.Sidekiq.enqueue_at/2` but not `ActiveJorb.QueueAdapter.Sidekiq.enqueue/1`.

While adding the job_id within the body of `ActiveJorb.QueueAdapter.Sidekiq.enqueue/1` is an easy fix, I wonder if there isn't a better place to put this that both `ActiveJorb.QueueAdapter.Sidekiq.enqueue/1` and `ActiveJorb.QueueAdapter.Sidekiq.enqueue_at/2` could make use of. AFAICT there isn't a way to have Elixir structs properties dynamically initialized by functions.

**Comment by jherdman** [comment]
@elucid a little bit better. Back at ya.

**Comment by elucid** [comment]
Thanks.

I think this is good for now--I've tested again. Feel free to merge and cut a release.

In the future, maybe we should consider using factory functions like Lance Halvorsen suggests in "Functional Web Development with Elixir, OTP, and Phoenix"

**Review by elucid — APPROVED** [review]

**Pull request merged**